lib/db_connection.ex

```diff
@@ -1445,73 +1378,67 @@ defmodule DBConnection do
     end
   end
 
-  defp fail(conn) do
-    case get_info(conn, nil) do
-      {:ok, conn_state, _meter} ->
-        put_info(conn, :failed, conn_state)
-      _ ->
-        :ok
+  defp fail(%DBConnection{pool_ref: pool_ref}) do
+    case Holder.status?(pool_ref, :ok) do
```

Comment:
The reason why I don't want to return the status is because there are two external status that we would have to handle:

1. the done status
2. the missing ets status

So having this as ok? and aborted? is better IMO but the current code is also fine, especially since we have put_status elsewhere.